### PR TITLE
JAXB and Mockito

### PR DESCRIPTION
**Pull request by un-install**

### CODE REVIEW

File: src/main/java/hw12/MateGroup.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	@XmlElementWrapper(name = "hrs")
+	@XmlElement(name = "humanResource")
+	private Set<HumanResource> humanResources;
+	@XmlElement(required = true, nillable = false, name = "speciality")
+	private String speciality;
+	@XmlElement(name = "maximal amount of students")
+	private int maxNumberOfStudents;
+	private int id;
+
+	public MateGroup(Person teacher, List<Person> students, Room room, Set<HumanResource> humanResources, String soeciality, int maxNumberOfStudents) {
```
Comment (DadizhaO) [inline src/main/java/hw12/MateGroup.java]: Code that we written on the lesson

File: src/test/java/hw12/StringChangeServiceUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+@RunWith(MockitoJUnitRunner.class)
+public class StringChangeServiceUnitTest {
+
+    @Spy
+    @InjectMocks
+    private StringChangerService scs;
+
+    @Mock
+    private StringLowerUpperService slus;
+
```
Comment (DadizhaO) [inline src/test/java/hw12/StringChangeServiceUnitTest.java]: Make more-readable variables

File: src/test/java/hw12/StringChangeServiceUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    //spy1
+    //failing
+    public void addStartAndEndTest() {
+        doReturn("some").when(scs).addStart(Mockito.anyString(), Mockito.anyString());
+
+        String expected = "some";
+        String actual = scs.addStartAndEnd("some", null, "some");
+
+        assertEquals(expected, actual);
+    }
```
Comment (DadizhaO) [inline src/test/java/hw12/StringChangeServiceUnitTest.java]: If test is failing you must to add expected = Some exception or add more information why this test is failing

File: src/main/java/hw12/MateGroup.java (hunk 1 above)
Comment (un-install) [reply src/main/java/hw12/MateGroup.java]: ther is some changes

File: src/test/java/hw12/StringChangeServiceUnitTest.java (hunk 2 above)
Comment (un-install) [reply src/test/java/hw12/StringChangeServiceUnitTest.java]: Ok, I ll pay atantion in future

File: src/test/java/hw12/StringChangeServiceUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        String expected = "someUPPER";
+        String actual = scs.addStartAndEndUpper("some", "some", "some");
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    //spy3
+    //failing
+    public void addStartAndEndTotalLowerSpyTest() {
```
Comment (SergeyPasko) [inline src/test/java/hw12/StringChangeServiceUnitTest.java]: You can cheak unsuccess scenarion if need, but all your test must be "green"
you can use assertNotEquals, for example

Comment (un-install) [reply src/test/java/hw12/StringChangeServiceUnitTest.java]: So, if there is a bag we should let it be?

**Pull request closed without merging**